### PR TITLE
Cart redesign (GCOM-1031)

### PR DESCRIPTION
**Pull request by Jessevdpoel**

### CODE REVIEW

File: packages/next-ui/ActionCard/ActionCard.tsx [hunk 1]
```diff
@@ -265,7 +266,7 @@ export function ActionCard(props: ActionCardProps) {
           width: '100%',
           justifyContent: 'space-between',
           alignContent: 'stretch',
-          alignItems: 'center',
+          alignItems: 'stretch',
         }}
```
Comment (bramvanderholst) [inline packages/next-ui/ActionCard/ActionCard.tsx:270]: @Jessevdpoel Heb je dit (en justifyContent: 'space-between' hieronder op regel 292) goed getest bij de bestaande implementaties van ActionCard?

File: examples/magento-graphcms/plugins/ConfigurableCartItemActionCard.tsx
Comment (bramvanderholst) [inline examples/magento-graphcms/plugins/ConfigurableCartItemActionCard.tsx:1]: Lege file? 🤔

File: packages/magento-cart-items/CartItem/CartItemActionCard.tsx [hunk 2]
```diff
@@ -53,9 +53,47 @@ export function CartItemActionCard(props: CartItemActionCardProps) {
           />
         )
       }
-      price={<Money {...prices?.row_total} />}
-      title={name}
-      secondaryAction={<UpdateItemQuantity uid={uid} quantity={quantity} sx={{ my: 1 }} />}
+      price={<Money {...prices?.row_total_including_tax} />}
```
Comment (bramvanderholst) [inline packages/magento-cart-items/CartItem/CartItemActionCard.tsx]: We should probably check if the `cartDisplayPricesInclTax` config is set

File: packages/magento-cart-items/Api/CartItem.graphql [hunk 3]
```diff
@@ -24,6 +24,9 @@ fragment CartItem on CartItemInterface @injectable {
     price {
       ...Money
     }
+    price_including_tax {
+      ...Money
+    }
```
Comment (bramvanderholst) [inline packages/magento-cart-items/Api/CartItem.graphql:29]: Can we use the `@include` directive based on the `cartDisplayPricesInclTax` config for this?

File: examples/magento-graphcms/pages/cart.tsx [hunk 4]
```diff
@@ -23,14 +22,17 @@ import {
...
 import { LayoutOverlay, LayoutOverlayProps } from '../components'
 import { graphqlSharedClient } from '../lib/graphql/graphqlSsrClient'
 
 type Props = Record<string, unknown>
 type GetPageStaticProps = GetStaticProps<LayoutOverlayProps, Props>
 
 function CartPage() {
+  const isMobile = useMediaQuery<Theme>((theme) => theme.breakpoints.down('sm'), {
+    defaultMatches: false,
+  })
```
Comment (bramvanderholst) [inline examples/magento-graphcms/pages/cart.tsx:36]: Shouldn't this be `.down('md')`?

File: packages/magento-cart-items/CartItem/CartItemActionCard.tsx [hunk 5]
```diff
@@ -84,10 +92,22 @@ export function CartItemActionCard(props: CartItemActionCardProps) {
...
             gap: theme.spacings.xxs,
           })}
         >
-          <UpdateItemQuantity uid={uid} quantity={quantity} sx={{ my: 1 }} />
+          <UpdateItemQuantity
+            uid={uid}
+            quantity={quantity}
+            sx={(theme) => ({
+              [theme.breakpoints.down('sm')]: {
+                width: '70px',
```
Comment (bramvanderholst) [inline packages/magento-cart-items/CartItem/CartItemActionCard.tsx:115]: Double digit numbers don't fit when the width is 70px. Why is this necessary? Without this the row seems to fit fine.

File: packages/next-ui/ActionCard/ActionCard.tsx (hunk 1 above)
Comment (Jessevdpoel) [reply packages/next-ui/ActionCard/ActionCard.tsx:270]: This does not affect existing implementations

File: examples/magento-graphcms/pages/cart.tsx [hunk 6]
```diff
@@ -79,29 +81,21 @@ function CartPage() {
...
-                    ConfigurableCartItem,
-                    DownloadableCartItem: CartItem,
-                    SimpleCartItem: CartItem,
-                    VirtualCartItem: CartItem,
-                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-                    // @ts-ignore GiftCardProduct is only available in Commerce
-                    GiftCardCartItem: CartItem,
+                  cart={data.cart}
+                  itemProps={{
+                    size: isMobile ? 'small' : 'large',
```
Comment (paales) [inline examples/magento-graphcms/pages/cart.tsx:88]: useMediaQuery is not allowed in the examples directory.

I think it would be a good idea to create two props on CartItems directly: `sizeSm` / `sizeMd` and use the useMediaQuery inside the component.

File: examples/magento-graphcms/pages/cart.tsx [hunk 7]
```diff
@@ -79,29 +81,21 @@ function CartPage() {
         <Container maxWidth='md'>
           <>
             {hasItems ? (
-              <Box sx={(theme) => ({ mt: theme.spacings.lg })}>
+              <>
                 <CartItems
```
Comment (paales) [inline examples/magento-graphcms/pages/cart.tsx:85]: Maybe we do want to have a bit of spacing on the top, seems a bit cramped on mobile at the moment

File: examples/magento-graphcms/pages/cart.tsx [hunk 8]
```diff
@@ -115,7 +109,15 @@ function CartPage() {
...
   Layout: LayoutOverlay,
-  layoutProps: { variantMd: 'bottom', variantSm: 'bottom' },
+  layoutProps: {
+    variantMd: 'right',
+    variantSm: 'bottom',
+    widthMd: '900px',
+    sizeMd: 'floating',
+    sizeSm: 'full',
+    justifyMd: 'start',
+    justifySm: 'stretch',
```
Comment (paales) [inline examples/magento-graphcms/pages/cart.tsx:119]: I don't think justifySm is necessary here.

File: packages/magento-cart-items/CartItem/CartItem.tsx [hunk 9]
```diff
@@ -91,11 +91,10 @@ export function CartItem(props: CartItemProps) {
       <Badge
         color='default'
         badgeContent={
-          <RemoveItemFromCartFab
+          <RemoveItemFromCart
```
Comment (paales) [inline packages/magento-cart-items/CartItem/CartItem.tsx:94]: Can we restore the original RemoveItemFromCartFab and keep them side-by-side?

File: packages/magento-cart-items/CartItems/CartItems.tsx [hunk 10]
```diff
@@ -1,24 +1,41 @@
...
-export type CartProps = { renderer: CartItemRenderer } & CartItemsFragment
+export type CartProps = ActionCardLayoutProps & {
+  cart?: CartItemsFragment | null
+  itemProps?: Omit<
+    CartItemActionCardProps,
+    'cartItem' | 'layout' | 'onClick' | 'disabled' | 'selected' | 'reset' | 'color'
+  >
+} & { size?: 'small' | 'medium' | 'large' }
 
 export function CartItems(props: CartProps) {
```
Comment (paales) [inline packages/magento-cart-items/CartItems/CartItems.tsx:13]: Can we perhaps create a completely new component CartItemsCards or something, so we can keep the old behavior for now? The old component can be marked as `@deprecated`.

File: packages/magento-cart-items/RemoveItemFromCart/RemoveItemFromCart.tsx [hunk 11]
```diff
@@ -1,37 +1,31 @@
...
 
-export function RemoveItemFromCartFab(props: RemoveItemFromCartProps) {
-  const {
-    product: {},
-    uid,
-    quantity,
-    prices,
-    ...formProps
-  } = props
+export function RemoveItemFromCart(props: RemoveItemFromCartProps) {
```
Comment (paales) [inline packages/magento-cart-items/RemoveItemFromCart/RemoveItemFromCart.tsx:12]: Restore the old RemoveItemFromCartFab and keep it in the codebase.

File: packages/magento-cart-items/CartItem/CartItemActionCard.tsx [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+      size={size}
+      after={filterNonNullableKeys(errors).map((error) => (
+        <Box sx={{ color: 'error.main', typography: 'caption' }} key={error.message}>
+          {error.message}
+        </Box>
+      ))}
+      {...rest}
+    />
+  )
+}
```
Comment (paales) [inline packages/magento-cart-items/CartItem/CartItemActionCard.tsx:144]: Can we remove the components to inside the components directory?

File: packages/magento-cart-items/components/ActionCartItem/ActionCartItem.tsx [hunk 13]
```diff
@@ -49,9 +49,7 @@ export function ActionCartItem(props: CartItemFragment & ActionCartItemProps) {
           <UpdateItemQuantity uid={uid} quantity={quantity} />
         </>
       }
-      action={
-        <RemoveItemFromCartFab uid={uid} quantity={quantity} prices={prices} product={product} />
-      }
+      action={<RemoveItemFromCart uid={uid} quantity={quantity} prices={prices} />}
```
Comment (paales) [inline packages/magento-cart-items/components/ActionCartItem/ActionCartItem.tsx:52]: Wasn't this the old incomplete item? Should we delete this component?

File: packages/magento-cart-items/components/SelectedCustomizableOptions/SelectedCustomizableOptions.tsx [hunk 14]
```diff
@@ -19,15 +19,30 @@ export function SelectedCustomizableOptions(props: ConfigurableActionCartItemPro
...
+                {value.price.value > 0 && <Money value={value.price.value} />} {value?.label}
+              </span>
+              <Box
+                sx={(theme) => ({
+                  [theme.breakpoints.down('sm')]: {
+                    display: 'none',
+                  },
+                })}
+              >
+                {value.value}
```
Comment (paales) [inline packages/magento-cart-items/components/SelectedCustomizableOptions/SelectedCustomizableOptions.tsx:43]: I don't think it is a good idea to hide stuff on mobile, can't we solve this in a different way? If information is important, it is also important on mobile

File: packages/magento-product-bundle/components/BundleCartItem/BundleCartItem.graphql [hunk 15]
```diff
@@ -1,4 +1,9 @@
 fragment BundleCartItem on BundleCartItem @inject(into: ["CartItem"]) {
+  prices {
+    price {
+      currency
+    }
+  }
```
Comment (paales) [inline packages/magento-product-bundle/components/BundleCartItem/BundleCartItem.graphql:6]: I don't think this something we'd want? Can you explain what you want to do with the currency field?

File: packages/next-ui/ActionCard/ActionCard.tsx (hunk 1 above)
Comment (paales) [reply packages/next-ui/ActionCard/ActionCard.tsx:270]: Do all shipping and payment steps still look as intended?

Comment (Jessevdpoel) [reply packages/next-ui/ActionCard/ActionCard.tsx:270]: Moved styling from ActionCard to CartItemActionCard

File: packages/ecommerce-ui/components/FormComponents/NumberFieldElement.tsx [hunk 16]
```diff
@@ -80,43 +82,69 @@ export function NumberFieldElement<T extends FieldValues>(props: NumberFieldElem
...
+                },
+              },
+              ...(Array.isArray(sx) ? sx : [sx]),
+            ]}
             autoComplete='off'
             InputProps={{
               ...textFieldProps.InputProps,
               startAdornment: (
-                <IconButton
+                <Fab
```
Comment (bramvanderholst) [inline packages/ecommerce-ui/components/FormComponents/NumberFieldElement.tsx:108]: What was the reason for switching to Fabs instead of IconButtons?
A Fab seems out of place in this context.

**Review by paales — CHANGES_REQUESTED** [review]

File: docs/components/CartItems.md [hunk 17]
```diff
@@ -0,0 +1,40 @@
+# Cart
```
[the remark this answers is not in the document]
Comment (paales) [reply docs/components/CartItems.md:1]: > If you are writing about old-to-new this should be in the release notes.
Or in the upgrade instructions.
